### PR TITLE
convert to primitive cell or to conventional cell

### PR DESCRIPTION
**Pull request by unkcpz**
Add to structure editor the cell convertor from or to primitive/conventional cell.

### CODE REVIEW

**Comment by unkcpz** [comment]
Hi @yakutovicha, can you have a look at this PR?

**Comment by unkcpz** [comment]
Tested on QeApp.

File: aiidalab_widgets_base/structures.py [hunk 1]
```diff
@@ -812,7 +775,7 @@ def _default_structure(self):
         return None
 
 
-def _register_operator(operator):
+def _register_structure(operator):
```
[the remark this answers is not in the document]
Comment (unkcpz) [reply aiidalab_widgets_base/structures.py:818]: True, I always have this bad habit to add something else 'by the way' without a clear independent PR. Will be careful in the future.

**Comment by unkcpz** [comment]
@yakutovicha I think this PR is ready.

File: aiidalab_widgets_base/structures.py [hunk 2]
```diff
@@ -842,15 +869,88 @@ def inner(ref, *args, **kwargs):
...
+        self.structure = atoms
+
+    @_register_structure
+    def def_conventional_cell(self, _=None, atoms=None):
+        atoms = self._to_standardize_cell(atoms, to_primitive=False)
+
+        self.structure = atoms
+
+    @staticmethod
+    def _to_standardize_cell(
```
Comment (yakutovicha) [inline aiidalab_widgets_base/structures.py:927]: I think it should be called `to_standard_cell` or `stardardize_cell`.

File: aiidalab_widgets_base/structures.py [hunk 3]
```diff
@@ -842,15 +869,88 @@ def inner(ref, *args, **kwargs):
...
+            layout={"width": "initial"},
+        )
+        conventional_cell.on_click(self.def_conventional_cell)
+
+        super().__init__(
+            children=[
+                ipw.HTML(
+                    "<b>Cell transformation:</b>",
+                    layout={"margin": "20px 0px 10px 0px"},
+                ),
```
Comment (yakutovicha) [inline aiidalab_widgets_base/structures.py:904]: I would argue this is a bit redundant. "Cell transformation" can also be set as a title.

**Review by yakutovicha — CHANGES_REQUESTED** [review]
Hi @unkcpz. I tested it, looks good. Just a few minor suggestions.

**Comment by superstar54** [comment]
`Cell Transformation` is a more general concept. If we use this term, it is better to include the general transformation matrix. Please have a look at `make_supercell` method here. https://wiki.fysik.dtu.dk/ase/ase/build/tools.html#ase.build.make_supercell
Also mentioned in #364

File: aiidalab_widgets_base/structures.py (hunk 2 above)
Comment (unkcpz) [reply aiidalab_widgets_base/structures.py]: Fair point, I use `to_standard_cell`.

File: aiidalab_widgets_base/structures.py (hunk 3 above)
Comment (unkcpz) [reply aiidalab_widgets_base/structures.py]: That's true, I use "Cell transform" for the default tab title and remove this HTML widget title.

**Comment by unkcpz** [comment]
@superstar54 Thanks for the advice. It is a very good suggestion, but I think it is better to be implemented it in a separate PR. Because it involves some design choice such as what we expect as the interface to input the transformation matrix that is worth having more discussion. I convert the item in your issue to a separate issue #365.

File: aiidalab_widgets_base/structures.py (hunk 2 above)
Comment (yakutovicha) [reply aiidalab_widgets_base/structures.py]: I think the function is now called `to_standardize_cell`. I am confused :)

Comment (unkcpz) [reply aiidalab_widgets_base/structures.py]: Ah, my bad. Can not focus on the so early train. I change it to `to_standard_cell`.

**Review by yakutovicha — APPROVED** [review]
LGTM!

**Pull request merged**